### PR TITLE
Redesiged the WasmRunner to allow 64bit parameters and return values.

### PR DESCRIPTION
**Pull request by gahaas**
The WasmRunner uses the existing WasmFunctionCompiler now to generate machine code for a given Wasm code. Thereby we can avoid using the GraphBuilderTester and we remove duplicated code which existed previously in WasmRunner and WasmFunctionCompiler. The generated code is then executed using the BufferedRawMachineAssemblerTester, which allows to pass 64bit parameters to the generated code.

This pull request depends on issue https://codereview.chromium.org/1414203009 in V8.

### CODE REVIEW

File: test/cctest/wasm/test-run-wasm.cc [hunk 1]
```diff
@@ -338,6 +272,137 @@ class WasmFunctionCompiler : public HandleAndZoneScope,
...
+  }
+
+
+  FunctionEnv* environment() { return &compiler_.env; }
+
+
+  // Builds a graph from the given Wasm code, and generates the machine
+  // code and call wrapper for that graph. This method must not be called
+  // more than once.
+  void Build(const byte* start, const byte* end) {
```
Comment (titzer) [inline test/cctest/wasm/test-run-wasm.cc]: Can you either enforce the restriction of a single call (with a DCHECK) or make it cache the result so that you can?

File: test/cctest/wasm/test-run-wasm.cc [hunk 2]
```diff
@@ -338,6 +272,137 @@ class WasmFunctionCompiler : public HandleAndZoneScope,
...
+      storage[index++] = LocalTypeFor(ret);
+    }
+    if (p0 != kMachNone) storage[index++] = LocalTypeFor(p0);
+    if (p1 != kMachNone) storage[index++] = LocalTypeFor(p1);
+    if (p2 != kMachNone) storage[index++] = LocalTypeFor(p2);
+    if (p3 != kMachNone) storage[index++] = LocalTypeFor(p3);
+  }
+
+
+  FunctionEnv* environment() { return &compiler_.env; }
```
Comment (titzer) [inline test/cctest/wasm/test-run-wasm.cc]: You can shorten this to env(); it's test code.

File: test/cctest/wasm/test-run-wasm.cc [hunk 3]
```diff
@@ -338,6 +272,137 @@ class WasmFunctionCompiler : public HandleAndZoneScope,
...
+  void Build(const byte* start, const byte* end) {
+    // Build the TF graph.
+    compiler_.Build(start, end);
+    // Generate code.
+    Handle<Code> code = compiler_.Compile(environment()->module);
+
+    // Construct the call wrapper.
+    if (signature_.parameter_count() == 0) {
+    call_wrapper_.Return(
+    call_wrapper_.AddNode(
```
Comment (titzer) [inline test/cctest/wasm/test-run-wasm.cc]: Can we use a loop here instead of a switch over the parameter count?

File: test/cctest/wasm/test-run-wasm.cc (hunk 1 above)
Comment (gahaas) [inline test/cctest/wasm/test-run-wasm.cc]: Done. I added the flag compilation_done_ which is checked with a DCHECK.

File: test/cctest/wasm/test-run-wasm.cc (hunk 2 above)
Comment (gahaas) [inline test/cctest/wasm/test-run-wasm.cc]: Done.

File: test/cctest/wasm/test-run-wasm.cc (hunk 3 above)
Comment (gahaas) [inline test/cctest/wasm/test-run-wasm.cc]: Done.

**Comment by titzer** [comment]
lgtm
